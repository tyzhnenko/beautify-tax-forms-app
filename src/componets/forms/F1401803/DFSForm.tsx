import { useState, useMemo } from "react";

import DFSFormUploadFile from "../DFSFormUploadFile";
import { DFSFormBody } from "./DFSFormBody";
import { DFSFormHeader } from "./DFSFormHeader";
import DFSXMLForm from "../../../utils/DFSXMLForm";
import { DFSFormHeaderType, DFSFormRowType } from "../../../utils/DFSXMLForm";
import { Alert } from "@material-tailwind/react";
import useTitle from "../../../utils/useTitle";

const defaultHeader = (): DFSFormHeaderType => ({
  firstName: "",
  lastName: "",
  sureName: "",
  taxNumber: "",
  taxAuthorityName: "",
  taxAuthorityNumber: "",
  email: "",
  periodFromYear: "",
  periodFromQuarter: "",
  periodToYear: "",
  periodToQuarter: "",
});

const defaultBody = (): Array<DFSFormRowType> => [
  {
    rowNumber: "",
    requestResult: "",
    quarterMonth: "",
    year: "",
    agentNumber: "",
    agentName: "",
    incomeCounted: "",
    incomePaid: "",
    taxCounted: "",
    taxPaid: "",
    taxCode: "",
    hiredDate: "",
    firedDate: "",
  },
];

function DFSForm() {
  useTitle("Форма F1401803");

  const [formXML, setFormXML] = useState<string>("");
  const [formHeader, setFormHeader] = useState<DFSFormHeaderType>(
    defaultHeader()
  );
  const [formBody, setFormBody] = useState<Array<DFSFormRowType>>(
    defaultBody()
  );
  const [errorMessage, setErrorMessage] = useState("");

  useMemo(() => {
    console.debug("formXML state is changed");
    if (formXML === "") {
      console.debug("formXML is empty");
      return;
    }
    const xmlForm = new DFSXMLForm(formXML);
    if (!xmlForm.validFormType()) {
      setErrorMessage("Invalid form type");
      return;
    }
    setFormHeader(xmlForm.getHeader());
    setFormBody(xmlForm.getBody());
  }, [formXML]);

  return (
    <>
      <Alert
        show={Boolean(errorMessage)}
        color="red"
        dismissible={{
          onClose: () => setErrorMessage(""),
        }}
      >
        {errorMessage}
      </Alert>
      <div className="shadow-lg">
        <DFSFormHeader headers={formHeader}></DFSFormHeader>
        <DFSFormBody rows={formBody}></DFSFormBody>
        <DFSFormUploadFile xmlLoader={setFormXML}></DFSFormUploadFile>
      </div>
    </>
  );
}

export default DFSForm;
