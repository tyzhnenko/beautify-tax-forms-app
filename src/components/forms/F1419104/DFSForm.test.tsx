/**
 * @jest-environment jsdom
 */

import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import DFSForm from "./DFSForm";
import fs from "fs";
import { TextEncoder, TextDecoder } from "util";

(global as any).TextEncoder = TextEncoder;
(global as any).TextDecoder = TextDecoder;

describe("DFSFormBody component", () => {
  function getMockXML() {
    const mockData = fs.readFileSync(
      "src/components/forms/F1419103/__testdata__/F1419103_mock_data.xml"
    );
    return new File([mockData], "test.xml", { type: "text/xml" });
  }

  it("should render the component", () => {
    render(<DFSForm />);

    expect(document.title).toEqual("Форма F1419103");
    expect(
      screen.getByText("Фізична особа - платник податків")
    ).toBeInTheDocument();
    expect(screen.getByText("№ з/п")).toBeInTheDocument();
  });

  it("should load the correct file", async () => {
    render(<DFSForm />);

    const mockFile = getMockXML();

    fireEvent.click(screen.getByRole("button", { name: /Оберить файл/i }));
    await screen.findByText(/Вибір файлу/i);

    const fileSelector = screen.getByTestId("file-selector");
    const uploader = screen.getByRole("button", { name: /Обробити/i });
    fireEvent.change(fileSelector, {
      target: { files: [mockFile] },
    });
    fireEvent.click(uploader);
    await screen.findByText(/user@example.com/i);

    const expectedTaxValues = [
      "10.1",
      "11.1",
      "12.1",
      "13.1",
      "14.1",
      "15.1",
      "16.1",
      "17.1",
      "18.1",
      "19.1",
      "20.1",
      "21.1",
    ];
    expectedTaxValues.forEach((value) => {
      expect(screen.getAllByText(value)).toHaveLength(2);
    });

    const expectedIncomeValues = [
      "100.00",
      "200.20",
      "300.30",
      "400.40",
      "500.50",
      "600.60",
      "700.70",
      "800.80",
      "900.90",
      "1000.00",
      "1100.10",
      "1200.20",
    ];
    expectedIncomeValues.forEach((value) => {
      expect(screen.getAllByText(value)).toHaveLength(2);
    });
  });

  it("should show an error if the file is not XML", async () => {
    render(<DFSForm />);

    const mockFile = new File(["test"], "test.txt", { type: "text/plain" });

    fireEvent.click(screen.getByRole("button", { name: /Оберить файл/i }));
    await screen.findByText(/Вибір файлу/i);

    const fileSelector = screen.getByTestId("file-selector");
    const uploader = screen.getByRole("button", { name: /Обробити/i });
    fireEvent.change(fileSelector, {
      target: { files: [mockFile] },
    });
    fireEvent.click(uploader);
    await screen.findByText("Не вірний тип форми");
  });

  it("should match the snapshot with no data", () => {
    const { container } = render(<DFSForm />);
    expect(container).toMatchSnapshot();
  });

  it("should match the snapshot with data", async () => {
    const { container } = render(<DFSForm />);
    const mockFile = getMockXML();

    fireEvent.click(screen.getByRole("button", { name: /Оберить файл/i }));
    await screen.findByText(/Вибір файлу/i);

    const fileSelector = screen.getByTestId("file-selector");
    const uploader = screen.getByRole("button", { name: /Обробити/i });
    fireEvent.change(fileSelector, {
      target: { files: [mockFile] },
    });
    fireEvent.click(uploader);
    await screen.findByText(/user@example.com/i);

    expect(container).toMatchSnapshot();
  });
});
