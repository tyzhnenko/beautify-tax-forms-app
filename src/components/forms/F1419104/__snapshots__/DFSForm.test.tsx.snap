// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DFSFormBody component should match the snapshot with data 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
>
  <div
    class="shadow-lg"
  >
    <div
      class="container"
    >
      <div
        class="grid grid-rows-10 grid-cols-12"
      >
        <div
          class="row-span-8 col-span-3 flex justify-center items-center border-2"
        >
          Фізична особа - платник податків
        </div>
        <div
          class="col-span-9 border-2 text-center"
        >
          пїЅпїЅ'пїЅ
        </div>
        <div
          class="col-span-9 border-2 text-center"
        >
          Прізвище
        </div>
        <div
          class="col-span-9 border-2 text-center"
        >
          пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ
        </div>
        <div
          class="col-span-9 border-2 text-center"
        >
          Ім'я
        </div>
        <div
          class="col-span-9 border-2 text-center"
        >
          пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅВІ
        </div>
        <div
          class="col-span-9 border-2 text-center"
        >
          По батькові
        </div>
        <div
          class="col-span-9 border-2 text-center"
        >
          1234567890
        </div>
        <div
          class="col-span-9 border-2 text-center"
        >
          Реестраційний номер облікової картки платника податків/серія (за наявності) та номер паспорта
        </div>
      </div>
      <div
        class="grid grid-rows-2 grid-cols-12"
      >
        <div
          class="row-span-2 col-span-3 flex justify-center items-center border-2"
        >
          Який зареєстрований в
        </div>
        <div
          class="col-span-1 border-2 text-center"
        >
          2034
        </div>
        <div
          class="col-span-8 border-2 text-center"
        >
          пїЅпїЅпїЅпїЅпїЅпїЅпїЅ пїЅпїЅпїЅпїЅпїЅЛІпїЅпїЅпїЅ пїЅпїЅпїЅ пїЅ пїЅпїЅпїЅКІпїЅпїЅпїЅКІпїЅ пїЅпїЅпїЅпїЅпїЅТІ,пїЅИЇпїЅпїЅпїЅпїЅпїЅ пїЅПІ(пїЅпїЅпїЅТІпїЅпїЅпїЅпїЅпїЅпїЅ пїЅпїЅпїЅпїЅпїЅ пїЅ.пїЅпїЅпїЅпїЅпїЅпїЅпїЅ)
        </div>
        <div
          class="col-span-1 border-2 text-center"
        >
          Код
        </div>
        <div
          class="col-span-8 border-2 text-center"
        >
          Назва териториального органу ДПС
        </div>
      </div>
      <div
        class="grid grid-rows-1 grid-cols-12"
      >
        <div
          class="col-span-3 text-center border-2"
        >
          Електронна адреса фізичної особи
        </div>
        <div
          class="col-span-9 text-center border-2"
        >
          user@example.com
        </div>
      </div>
      <div
        class="flex pt-4"
      >
        <div
          class="flex-initial w-8"
        />
        <div
          class="flex-auto w-8 text-right"
        >
          за период з
        </div>
        <div
          class="flex-auto w-8 text-center border-b-2 border-black"
        >
          1
        </div>
        <div
          class="flex-auto w-8 text-left"
        >
          кварталу
        </div>
        <div
          class="flex-auto w-8 text-center border-b-2 border-black"
        >
          2022
        </div>
        <div
          class="flex-auto w-8"
        >
          року
        </div>
        <div
          class="w-32"
        />
        <div
          class="flex-auto w-8 text-right"
        >
          по
        </div>
        <div
          class="flex-auto w-8 text-center border-b-2 border-black"
        >
          4
        </div>
        <div
          class="flex-auto w-8 text-left"
        >
          квартал
        </div>
        <div
          class="flex-auto w-8 text-center border-b-2 border-black"
        >
          2022
        </div>
        <div
          class="flex-auto w-8"
        >
          року
        </div>
        <div
          class="w-8"
        />
      </div>
    </div>
    <div
      class="container overflow-hidden"
    >
      <table
        class="min-w-full text-center text-sm font-light border-2"
      >
        <thead
          class="font-medium dark:border-neutral-500"
        >
          <tr
            class="border-2"
          >
            <th
              class="border-2"
              rowspan="2"
              scope="col"
            >
              № з/п
            </th>
            <th
              class="border-2"
              rowspan="2"
              scope="col"
            >
              Результат обробки запиту
            </th>
            <th
              class="border-2"
              colspan="2"
              scope="col"
            >
              Період, за який надається інформація
            </th>
            <th
              class="border-2"
              colspan="2"
              scope="col"
            >
              Джерела отримання інформації
            </th>
            <th
              class="border-2"
              colspan="2"
              scope="col"
            >
              Сума доходу
            </th>
            <th
              class="border-2"
              colspan="2"
              scope="col"
            >
              Сума податку
            </th>
            <th
              class="border-2"
              colspan="2"
              scope="col"
            >
              Сума війскового збору
            </th>
            <th
              class="border-2"
              rowspan="2"
              scope="col"
            >
              Код та назва ознаку доходу / код ознаки пільги
            </th>
            <th
              class="border-2"
              rowspan="2"
              scope="col"
            >
              Дата прийому на роботу
            </th>
            <th
              class="border-2"
              rowspan="2"
              scope="col"
            >
              Дата звільненя з роботи
            </th>
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="border-2"
              scope="col"
            >
              Номер кварталу - місяць
            </th>
            <th
              class="whitespace-nowrap border-2"
              scope="col"
            >
              Рік
            </th>
            <th
              class="border-2"
              scope="col"
            >
              Податковий номер/серія(за наявності) та номер паспорта податкового агента
            </th>
            <th
              class="border-2"
              scope="col"
            >
              Найменування юридичної особи/ПІБ фізичної особи – підприємця
            </th>
            <th
              class="-rotate-90 border-2"
              scope="col"
            >
              нарахованого
            </th>
            <th
              class="-rotate-90 border-2"
              scope="col"
            >
              виплаченого
            </th>
            <th
              class="-rotate-90 border-2"
              scope="col"
            >
              нарахованого
            </th>
            <th
              class="-rotate-90 border-2"
              scope="col"
            >
              виплаченого
            </th>
            <th
              class="-rotate-90 border-2"
              scope="col"
            >
              нарахованого
            </th>
            <th
              class="-rotate-90 border-2"
              scope="col"
            >
              виплаченого
            </th>
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              3
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              4
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              5
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              6
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              7
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              8
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              9
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              10
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              11
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              12
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              13
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              14
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              15
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="border-2"
            >
              1 - СіпїЅпїЅпїЅпїЅ
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2022
            </th>
            <th
              class="border-2"
            >
              99999981
            </th>
            <th
              class="w-64 border-2"
            >
              пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ '99999981'
            </th>
            <th
              class="border-2"
            >
              100.00
            </th>
            <th
              class="border-2"
            >
              100.00
            </th>
            <th
              class="border-2"
            >
              10.1
            </th>
            <th
              class="border-2"
            >
              10.1
            </th>
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              2
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="border-2"
            >
              1 - СіпїЅпїЅпїЅпїЅ
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2022
            </th>
            <th
              class="border-2"
            >
              99999982
            </th>
            <th
              class="w-64 border-2"
            >
              пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ '99999982'
            </th>
            <th
              class="border-2"
            >
              200.20
            </th>
            <th
              class="border-2"
            >
              200.20
            </th>
            <th
              class="border-2"
            >
              11.1
            </th>
            <th
              class="border-2"
            >
              11.1
            </th>
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              3
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="border-2"
            >
              1 - СіпїЅпїЅпїЅпїЅ
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2022
            </th>
            <th
              class="border-2"
            >
              99999983
            </th>
            <th
              class="w-64 border-2"
            >
              пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ '99999983'
            </th>
            <th
              class="border-2"
            >
              300.30
            </th>
            <th
              class="border-2"
            >
              300.30
            </th>
            <th
              class="border-2"
            >
              12.1
            </th>
            <th
              class="border-2"
            >
              12.1
            </th>
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              4
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="border-2"
            >
              2 - пїЅпїЅпїЅпїЅпїЅ
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2022
            </th>
            <th
              class="border-2"
            >
              99999984
            </th>
            <th
              class="w-64 border-2"
            >
              пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ '99999984'
            </th>
            <th
              class="border-2"
            >
              400.40
            </th>
            <th
              class="border-2"
            >
              400.40
            </th>
            <th
              class="border-2"
            >
              13.1
            </th>
            <th
              class="border-2"
            >
              13.1
            </th>
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              5
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="border-2"
            >
              2 - пїЅпїЅпїЅпїЅпїЅ
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2022
            </th>
            <th
              class="border-2"
            >
              99999985
            </th>
            <th
              class="w-64 border-2"
            >
              пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ '99999985'
            </th>
            <th
              class="border-2"
            >
              500.50
            </th>
            <th
              class="border-2"
            >
              500.50
            </th>
            <th
              class="border-2"
            >
              14.1
            </th>
            <th
              class="border-2"
            >
              14.1
            </th>
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              6
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="border-2"
            >
              2 - пїЅпїЅпїЅпїЅпїЅ
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2022
            </th>
            <th
              class="border-2"
            >
              99999986
            </th>
            <th
              class="w-64 border-2"
            >
              пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ '99999986'
            </th>
            <th
              class="border-2"
            >
              600.60
            </th>
            <th
              class="border-2"
            >
              600.60
            </th>
            <th
              class="border-2"
            >
              15.1
            </th>
            <th
              class="border-2"
            >
              15.1
            </th>
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              7
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="border-2"
            >
              3 - пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2022
            </th>
            <th
              class="border-2"
            >
              99999987
            </th>
            <th
              class="w-64 border-2"
            >
              пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ '99999987'
            </th>
            <th
              class="border-2"
            >
              700.70
            </th>
            <th
              class="border-2"
            >
              700.70
            </th>
            <th
              class="border-2"
            >
              16.1
            </th>
            <th
              class="border-2"
            >
              16.1
            </th>
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              8
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="border-2"
            >
              3 - пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2022
            </th>
            <th
              class="border-2"
            >
              99999988
            </th>
            <th
              class="w-64 border-2"
            >
              пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ '99999988'
            </th>
            <th
              class="border-2"
            >
              800.80
            </th>
            <th
              class="border-2"
            >
              800.80
            </th>
            <th
              class="border-2"
            >
              17.1
            </th>
            <th
              class="border-2"
            >
              17.1
            </th>
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              9
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="border-2"
            >
              3 - пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2022
            </th>
            <th
              class="border-2"
            >
              99999989
            </th>
            <th
              class="w-64 border-2"
            >
              пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ '99999989'
            </th>
            <th
              class="border-2"
            >
              900.90
            </th>
            <th
              class="border-2"
            >
              900.90
            </th>
            <th
              class="border-2"
            >
              18.1
            </th>
            <th
              class="border-2"
            >
              18.1
            </th>
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              10
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="border-2"
            >
              4 - пїЅпїЅпїЅпїЅпїЅпїЅ
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2022
            </th>
            <th
              class="border-2"
            >
              99999990
            </th>
            <th
              class="w-64 border-2"
            >
              пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ '99999990'
            </th>
            <th
              class="border-2"
            >
              1000.00
            </th>
            <th
              class="border-2"
            >
              1000.00
            </th>
            <th
              class="border-2"
            >
              19.1
            </th>
            <th
              class="border-2"
            >
              19.1
            </th>
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              11
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="border-2"
            >
              4 - пїЅпїЅпїЅпїЅпїЅпїЅ
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2022
            </th>
            <th
              class="border-2"
            >
              99999991
            </th>
            <th
              class="w-64 border-2"
            >
              пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ '99999991'
            </th>
            <th
              class="border-2"
            >
              1100.10
            </th>
            <th
              class="border-2"
            >
              1100.10
            </th>
            <th
              class="border-2"
            >
              20.1
            </th>
            <th
              class="border-2"
            >
              20.1
            </th>
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              12
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2
            </th>
            <th
              class="border-2"
            >
              4
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2022
            </th>
            <th
              class="border-2"
            >
              1234567890
            </th>
            <th
              class="w-64 border-2"
            >
              пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅ пїЅпїЅ'пїЅ пїЅпїЅпїЅпїЅпїЅпїЅпїЅпїЅВІ
            </th>
            <th
              class="border-2"
            >
              1200.20
            </th>
            <th
              class="border-2"
            >
              1200.20
            </th>
            <th
              class="border-2"
            >
              21.1
            </th>
            <th
              class="border-2"
            >
              21.1
            </th>
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="text-center"
    >
      <button
        class="align-middle select-none font-sans font-bold text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs py-3 px-6 rounded-lg bg-gray-900 text-white shadow-md shadow-gray-900/10 hover:shadow-lg hover:shadow-gray-900/20 focus:opacity-[0.85] focus:shadow-none active:opacity-[0.85] active:shadow-none"
        type="button"
      >
        <span>
          Оберить файл
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`DFSFormBody component should match the snapshot with no data 1`] = `
<div>
  <div
    class="shadow-lg"
  >
    <div
      class="container"
    >
      <div
        class="grid grid-rows-10 grid-cols-12"
      >
        <div
          class="row-span-8 col-span-3 flex justify-center items-center border-2"
        >
          Фізична особа - платник податків
        </div>
        <div
          class="col-span-9 border-2 text-center"
        />
        <div
          class="col-span-9 border-2 text-center"
        >
          Прізвище
        </div>
        <div
          class="col-span-9 border-2 text-center"
        />
        <div
          class="col-span-9 border-2 text-center"
        >
          Ім'я
        </div>
        <div
          class="col-span-9 border-2 text-center"
        />
        <div
          class="col-span-9 border-2 text-center"
        >
          По батькові
        </div>
        <div
          class="col-span-9 border-2 text-center"
        />
        <div
          class="col-span-9 border-2 text-center"
        >
          Реестраційний номер облікової картки платника податків/серія (за наявності) та номер паспорта
        </div>
      </div>
      <div
        class="grid grid-rows-2 grid-cols-12"
      >
        <div
          class="row-span-2 col-span-3 flex justify-center items-center border-2"
        >
          Який зареєстрований в
        </div>
        <div
          class="col-span-1 border-2 text-center"
        />
        <div
          class="col-span-8 border-2 text-center"
        />
        <div
          class="col-span-1 border-2 text-center"
        >
          Код
        </div>
        <div
          class="col-span-8 border-2 text-center"
        >
          Назва териториального органу ДПС
        </div>
      </div>
      <div
        class="grid grid-rows-1 grid-cols-12"
      >
        <div
          class="col-span-3 text-center border-2"
        >
          Електронна адреса фізичної особи
        </div>
        <div
          class="col-span-9 text-center border-2"
        />
      </div>
      <div
        class="flex pt-4"
      >
        <div
          class="flex-initial w-8"
        />
        <div
          class="flex-auto w-8 text-right"
        >
          за период з
        </div>
        <div
          class="flex-auto w-8 text-center border-b-2 border-black"
        />
        <div
          class="flex-auto w-8 text-left"
        >
          кварталу
        </div>
        <div
          class="flex-auto w-8 text-center border-b-2 border-black"
        />
        <div
          class="flex-auto w-8"
        >
          року
        </div>
        <div
          class="w-32"
        />
        <div
          class="flex-auto w-8 text-right"
        >
          по
        </div>
        <div
          class="flex-auto w-8 text-center border-b-2 border-black"
        />
        <div
          class="flex-auto w-8 text-left"
        >
          квартал
        </div>
        <div
          class="flex-auto w-8 text-center border-b-2 border-black"
        />
        <div
          class="flex-auto w-8"
        >
          року
        </div>
        <div
          class="w-8"
        />
      </div>
    </div>
    <div
      class="container overflow-hidden"
    >
      <table
        class="min-w-full text-center text-sm font-light border-2"
      >
        <thead
          class="font-medium dark:border-neutral-500"
        >
          <tr
            class="border-2"
          >
            <th
              class="border-2"
              rowspan="2"
              scope="col"
            >
              № з/п
            </th>
            <th
              class="border-2"
              rowspan="2"
              scope="col"
            >
              Результат обробки запиту
            </th>
            <th
              class="border-2"
              colspan="2"
              scope="col"
            >
              Період, за який надається інформація
            </th>
            <th
              class="border-2"
              colspan="2"
              scope="col"
            >
              Джерела отримання інформації
            </th>
            <th
              class="border-2"
              colspan="2"
              scope="col"
            >
              Сума доходу
            </th>
            <th
              class="border-2"
              colspan="2"
              scope="col"
            >
              Сума податку
            </th>
            <th
              class="border-2"
              colspan="2"
              scope="col"
            >
              Сума війскового збору
            </th>
            <th
              class="border-2"
              rowspan="2"
              scope="col"
            >
              Код та назва ознаку доходу / код ознаки пільги
            </th>
            <th
              class="border-2"
              rowspan="2"
              scope="col"
            >
              Дата прийому на роботу
            </th>
            <th
              class="border-2"
              rowspan="2"
              scope="col"
            >
              Дата звільненя з роботи
            </th>
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="border-2"
              scope="col"
            >
              Номер кварталу - місяць
            </th>
            <th
              class="whitespace-nowrap border-2"
              scope="col"
            >
              Рік
            </th>
            <th
              class="border-2"
              scope="col"
            >
              Податковий номер/серія(за наявності) та номер паспорта податкового агента
            </th>
            <th
              class="border-2"
              scope="col"
            >
              Найменування юридичної особи/ПІБ фізичної особи – підприємця
            </th>
            <th
              class="-rotate-90 border-2"
              scope="col"
            >
              нарахованого
            </th>
            <th
              class="-rotate-90 border-2"
              scope="col"
            >
              виплаченого
            </th>
            <th
              class="-rotate-90 border-2"
              scope="col"
            >
              нарахованого
            </th>
            <th
              class="-rotate-90 border-2"
              scope="col"
            >
              виплаченого
            </th>
            <th
              class="-rotate-90 border-2"
              scope="col"
            >
              нарахованого
            </th>
            <th
              class="-rotate-90 border-2"
              scope="col"
            >
              виплаченого
            </th>
          </tr>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            >
              1
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              2
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              3
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              4
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              5
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              6
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              7
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              8
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              9
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              10
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              11
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              12
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              13
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              14
            </th>
            <th
              class="whitespace-nowrap border-2"
            >
              15
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="border-2"
          >
            <th
              class="whitespace-nowrap border-2"
            />
            <th
              class="whitespace-nowrap border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="whitespace-nowrap border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="w-64 border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
            <th
              class="border-2"
            />
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="text-center"
    >
      <button
        class="align-middle select-none font-sans font-bold text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs py-3 px-6 rounded-lg bg-gray-900 text-white shadow-md shadow-gray-900/10 hover:shadow-lg hover:shadow-gray-900/20 focus:opacity-[0.85] focus:shadow-none active:opacity-[0.85] active:shadow-none"
        type="button"
      >
        <span>
          Оберить файл
        </span>
      </button>
    </div>
  </div>
</div>
`;
